packages/net/tcp_connection.pony

```diff
@@ -955,3 +961,108 @@ actor TCPConnection
       _throttled = false
       _notify.unthrottled(this)
     end
+
+  /**************************************/
+
+  fun ref getsockopt(level: I32, option_name: I32, option: Array[U8]): (U32, U32) =>
+    """
+    General wrapper for TCP sockets to the `getsockopt(2)` system call.
+
+    In case of system call success, this function returns the 2-tuple:
+    1. The integer `0`.
+    2. The value of the `*(uint32_t)option_length` argument set by
+       `getsockopt(2)`.  The caller must use this length to properly
+        interpret the bytes written by the kernel into the `option`
+        byte array: this length may be smaller than `option`'s
+        original size.
+
+    In case of system call failure, this function returns the 2-tuple:
+    1. The value of `errno`.
+    2. An undefined value that must be ignored.
```

Comment:
A usage example would be nice.

Also the information that the array should be pre allocated the expected result size is missing.